MoviesProject/Home/HomeReducer.swift

```diff
@@ -22,38 +22,30 @@ public struct Home: Reducer {
                         await send(.moviesFetched(movies))
                     }
                 }
-            case .topList, .categoriesList:
+            case .topList, .categoriesList, .movieDetails, .movieSearch:
                 break
             case .moviesFetched(let movies):
                 if let movies {
                     state.moviesLoadingValue = .loaded(movies)
-                    state.topList.movies = movies
+                    
                     state.categoriesList.movies = movies
+                    state.movieSearch.movies = movies
                     
                     let sortedMovies = movies.sorted { $0.imdbRating > $1.imdbRating }
                     state.topList.sortedMovies = sortedMovies
                     state.categoriesList.sortedMovies = sortedMovies
-                    state.dataFetched = true
-                    
                     
                 } else {
                     state.moviesLoadingValue = .error("Failed fetching movies")
                 }
-            case .movieDetails:
-                break
-//            case .path(_):
-//                    .none
             case .searchBarTapped(let item):
                 print(item)
                 break
-            case .movieSearch:
-                break
+            case .searchTextDidChange(let txt):
+                state.searchText = txt
+                return .none
```

Comment:
you dont have to return .none explicitly here. It will be returned either way after the switch statement